python/src/lib.rs

```
@@ -331,6 +335,29 @@ impl RawDeltaTable {
         Ok(serde_json::to_string(&metrics).unwrap())
     }
 
+
+    #[pyo3(signature = ())]
+    pub fn commit_optimize(
+        &mut self,
+    ) -> PyResult<String> {
+        let cmd = OptimizeBuilder::new(
+            self._table.log_store(),
+            self._table.snapshot().map_err(PythonError::from)?.clone(),
+        )
+        .with_max_concurrent_tasks(num_cpus::get());
+
+        let commit = match &self._commit {
+            Some(c) => c,
+            None => panic!("No commit found from previous optimize, cannot commit."),
```

should throw a python exception instead.